lib/proxy/helpers.rb

```diff
@@ -7,16 +7,18 @@ module Proxy::Helpers
   # Accepts a html error code and a message, which is then returned to the caller after adding to the proxy log
   # OR  a block which is executed and its errors handled in a similar way.
   # If no code is supplied when the block is declared then the html error used is 400.
-  def log_halt code=nil, exception=nil
-    message = exception.is_a?(String) ? exception : exception.to_s
+  def log_halt code=nil, exception_or_msg=nil, custom_msg=nil
+    message = exception_or_msg.is_a?(String) ? exception_or_msg : exception_or_msg.to_s
+    message = "#{custom_msg}: #{message}" if custom_msg
+    exception = exception_or_msg.is_a?(Exception) ? exception_or_msg.message : exception_or_msg
```

Comment:
This here breaks line 25 since it means `exception.is?(Exception)` will always be false meaning you no longer get backtraces.